### ISSUE TITLE
BUG during call :  microphone and speaker at the same time leads to cut out

### ISSUE THREAD

**Issue by philrandom** [hselasky/virtual_oss#11]
Hello,
On FreeBSD 12.1 I encounter this issue :
- In a call Microsoft Teams via **Chromium**  the microphone cuts out (stop working) about a minute after the beginning of a convo
- In a call via **telegram-desktop** microphone is working but speaker isn't

the cut ends when the application stops to use `/dev/dsp`.
### Context
Following this [documentation of FreeBSD](https://wiki.freebsd.org/Sound#virtual_oss_.28advanced.29)
```
cat /dev/sndstat
-----------------------------------------------------------------------------------------------
Installed devices:
pcm0: <Intel Haswell (HDMI/DP 8ch)> (play)
pcm1: <Realtek ALC292 (Analog 2.0+HP/2.0)> (play/rec) default
pcm2: <Realtek ALC292 (Internal Analog Mic)> (rec)
Installed devices from userspace:
dsp: <Virtual OSS> (play/rec)
``` 
```
cat /usr/local/etc/rc.d/virtual_oss
------------------------------------------------------------------------------------------------
virtual_oss_default_args="\
  -T /dev/sndstat \
  -S \
  -i 8 \
  -C 2 -c 2 \
  -r 48000 \
  -b 24 \
  -s 8.0ms \
  -O /dev/dsp1 \
  -R /dev/dsp2 \
  -c 2 \
  -d dsp \
  -t dsp.ctl"
```
### Thoughts
- First of all I'm not sure where the problem is, if it's in `cuse` or `sndiod` or `virtual_oss`
- I suspect that the problem can come from :
  - buffer overflow
  - `poll(2)` synchronization in the function `oss_rec_transfert()` in `backend_oss/backend_oss.c`
  - miss-configured virtual_oss (maybe channel)

**Comment by hselasky** [comment]
Can you check how many dsp devices are opened, and that there is no leak of opened devices?

**Comment by hselasky** [comment]
Also check "dmesg" for any errors.

**Comment by hselasky** [comment]
Can you check the CPU usage in "top" when this happens. Some chrome audio applications use insanely small buffers, which cause sometimes too many system calls for OSS to handle.

**Comment by hselasky** [comment]
Ping - can you check version 1.2.7 of virtual_oss ?

**Comment by philrandom** [comment]
@hselasky excuse me I haven't got enough time. So I did an update from FreeBSD 12.1 to 12.2 and also **update virtual_oss from 1.2.6 to 1.2.7**. I kept the same parameters in `/usr/local/etc/rc.d/virtual_oss`.  Calls on Teams via Chromium works perfect !! The quality is far better than on archlinux (my previous os in 2019) with alsa. 

But `telegram-desktop` isn't able to connect on microphone, the problem come from a bad configuration of *PulseAudio*, my bad.

**Thank you and development team for your time, and for the quality of virtual_oss**
